server/plugin.go

```diff
@@ -142,6 +142,22 @@ func (p *Plugin) GetClientForUser(userID string) (msteams.Client, error) {
 	if token == nil {
 		return nil, errors.New("not connected user")
 	}
+
+	if token.Expiry.Before(time.Now()) {
+		var err error
+		token, err = msteams.RefreshToken(p.GetURL()+"/oauth-redirect", p.getConfiguration().TenantID, p.getConfiguration().ClientID, p.getConfiguration().ClientSecret, token)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
I wonder if we can move this to a `client.Refresh`? Or does the client get "used" by the builder before we can refresh it ourselves?

Reply:
I can check it, but my concern is that the whole process of creating the client can lead to refreshing the token without noticing, but let me check it, and I'll see if we can move that as a method.

Reply:
Done